### PR TITLE
loads action handler for a group of modules

### PR DESCRIPTION
**Pull request by privateip**
* loads action handler based on module group prefix

### CODE REVIEW

File: lib/ansible/executor/task_executor.py [hunk 1]
```diff
@@ -758,9 +758,14 @@ def _get_action_handler(self, connection, templar):
         Returns the correct action plugin to handle the requestion task action
         '''
 
+        network_group_modules = frozenset(['eos', 'nxos', 'ios', 'iosxr', 'junos', 'vyos'])
```
Comment (nitzmahone) [inline lib/ansible/executor/task_executor.py:761]: Eek, having this list hardcoded in the task_executor feels wrong... I'm trying to think of a sane way to do it that wouldn't require actually loading all the action plugins to have them register an intent like this (and that also wouldn't open up the potential for "catchall" action plugins that could hijack things).

File: lib/ansible/executor/task_executor.py [hunk 2]
```diff
@@ -758,9 +758,14 @@ def _get_action_handler(self, connection, templar):
...
+        module_prefix = self._task.action.split('_')[0]
+
         # let action plugin override module, fallback to 'normal' action plugin otherwise
         if self._task.action in self._shared_loader_obj.action_loader:
             handler_name = self._task.action
+        elif all((module_prefix in network_group_modules, module_prefix in self._shared_loader_obj.action_loader)):
```
Comment (nitzmahone) [inline lib/ansible/executor/task_executor.py:767]: Why the all() instead of just an and?

File: lib/ansible/executor/task_executor.py (hunk 1 above)
Comment (privateip) [reply lib/ansible/executor/task_executor.py:761]: Given where we are in the dev cycle, this gets us through 2.3 and then we need to commit to finding a better long term solution.  The drawback to not doing this is that every single network module will need a local action in 2.3

File: lib/ansible/executor/task_executor.py (hunk 2 above)
Comment (privateip) [reply lib/ansible/executor/task_executor.py:767]: module_prefix is generated by splitting on `_` so we need to be careful to not hit a false positive here

**Pull request merged**